VERSIONING.md

```diff
@@ -0,0 +1,104 @@
+and lists public API in the directory "publicAPI". Any changes to public API,
+without corresponding changes here will result in build breaks - this helps
+catch any unintended changes to public API from being shipped accidentally. This
+also helps reviewers quickly understand if a given PR is making public API
+changes. For example,
+[this](https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/src/OpenTelemetry.Instrumentation.AspNetCore/.publicApi)
+shows the public per target framework for the
+`OpenTelemetry.Instrumentation.AspNetCore` package.
+
+Since no stable version has been released so far, every API is listed in the
+"Unshipped.txt" file. Once 1.0.0 is shipped, it'll be moved to "Shipped.txt"
+file.
+
+## Packaging
+
+OpenTelemetry is structured around signals like Traces, Metrics, Logs, Baggage
+etc. OpenTelemetry .NET does not ship separate packages per signal. There is a
+single package which includes all the signals. i.e `OpenTelemetry.API` package
+will consist of API components from *all* the signals. `OpenTelemetry` package
+will consist of SDK components from *all* the signals. Instrumentations also
+follow the same model - for example, There will be a single package
+`OpenTelemetry.Instrumentation.AspNetCore` for ASP.NET Core instrumentation,
+which produces Traces, Metrics and handles propagation, instead of separate
+packages for traces and metrics.
+
+## Experimental Signals
+
+### Current approach
+
+Given the fact that all signals are shipped as a single package, the following
+approach is currently used to deal with experimental signals.
+
+Any experimental signal which is shipped as part of a normal package will be
+marked
+[Obsolete](https://docs.microsoft.com/dotnet/api/system.obsoleteattribute).
+(Ideally we need an "Experimental" attribute, but .NET has only built-in support
+for "Obsolete", so we are leveraging it. Using "Obsolete" code results in
+compile time warnings to caution users.) This allows an experimental signal to
+co-exist in the same package as other, non-experimental signals. For example,
+the OpenTelemetry 1.0.0 package will consist of Traces and Metrics. As Metrics
+signal is not ready for stable release, every method dealing with Metrics will
+be marked "Obsolete". Once Metrics signal gets stable, it'll be added to a
+release with minor version bump, say 1.2.0. The "Obsolete" Metric methods can be
+removed with a major version bump.
+
+### Alternate option
+
+An alternate option to deal with experimental signal is to remove the signal
+from the common packages, and ship as separate package. For example,
+OpenTelemetry 1.0.0 will only contain Traces, and there will be a separate
+package OpenTelemetry.Metrics.Experimental, which contains the metric signal.
+Once the signal achieve stable quality, it'll be made part of the main package
+and released as a minor version update. i.e OpenTelemetry 1.2.0 will contain
+Traces and Metrics.
+
+## Examples
```

Comment:
Probably adding a section with examples using separate packages would be nice, to be able to put things into perspective for such approach.